src/rpcmining.cpp

```diff
@@ -605,14 +635,30 @@ Value getblocktemplate(const Array& params, bool fHelp)
     result.push_back(Pair("capabilities", aCaps));
     result.push_back(Pair("version", pblock->nVersion));
     result.push_back(Pair("previousblockhash", pblock->hashPrevBlock.GetHex()));
+    result.push_back(Pair("reserved", pblock->hashReserved.GetHex()));
     result.push_back(Pair("transactions", transactions));
-    result.push_back(Pair("coinbaseaux", aux));
-    result.push_back(Pair("coinbasevalue", (int64_t)pblock->vtx[0].vout[0].nValue));
+    if (coinbasetxn) {
+        assert(txCoinbase.type() == obj_type);
+        result.push_back(Pair("coinbasetxn", txCoinbase));
+    } else {
+        result.push_back(Pair("coinbaseaux", aux));
+        // Correct these if GetBlockTemplate changes the order
+        result.push_back(Pair("coinbasevalue", (int64_t)pblock->vtx[0].vout[0].nValue));
+        // Include founders' reward if it is required
+        if (pblock->vtx[0].vout.size() > 1) {
+            result.push_back(Pair("coinbasefrvalue",
+                                  (int64_t)pblock->vtx[0].vout[1].nValue));
+            result.push_back(Pair("coinbasefrscript",
+                                  HexStr(pblock->vtx[0].vout[1].scriptPubKey.begin(),
+                                         pblock->vtx[0].vout[1].scriptPubKey.end())
+                                 ));
+        }
+    }
     result.push_back(Pair("longpollid", chainActive.Tip()->GetBlockHash().GetHex() + i64tostr(nTransactionsUpdatedLast)));
     result.push_back(Pair("target", hashTarget.GetHex()));
     result.push_back(Pair("mintime", (int64_t)pindexPrev->GetMedianTimePast()+1));
     result.push_back(Pair("mutable", aMutable));
```

Comment:
Isn't it implied if `coinbasetxn` is omitted? I assume you mean we should have it for `coinbasetxn` too, but that would require either specifying how the miner can edit the coinbase while maintaining the Founders' Reward, or specifying the FR script for `coinbasetxn` too. I'm inclined to just leave it as-is, keeping the changes minimal in scope until we have a chance to properly specify them.